server/src/test/java/org/elasticsearch/snapshots/SnapshotsServiceTests.java

```diff
@@ -346,12 +421,24 @@ protected ConnectionStatus getConnectionStatus(DiscoveryNode sender, DiscoveryNo
 
                 @Override
                 protected Optional<DisruptableMockTransport> getDisruptedCapturingTransport(DiscoveryNode node, String action) {
-                    return Optional.ofNullable(testClusterNodes.nodes.get(node.getName()).mockTransport);
+                    final Predicate<TestClusterNode> matchesDestination;
```

Comment:
The logic in this override is of the disruptable mock transport is now super similar to what `CoordinatorTests` use, we can probably just dry things up here in a next step.

Reply:
++